### ISSUE TITLE
Sale Order shows "Not Shippable" items when Shipment is "in process"

### ISSUE THREAD

**Issue by jrgonzalezrios** [Dolibarr/dolibarr#13511]
# Bug
When listing all Sale Orders, a red icon shows with a "not shippable" label. This red icon/label should not appear when a shipment exists as "in process". Items are not in the inventory, which is true, but they are not in the inventory anymore because of the shipment.

## Environment
- Version: 11.0.3
- OS: Linux
- Web server: Nginx/1.15.8
- PHP: 7.3.9
- Database: MySql 7.7.27

## Expected and actual behavior
When a shipment exists as "In process" it should not show a "shippable" nor "not shippable" icon/label. It should change to "In process" since the items are not in the inventory because of shipment itself.

## Steps to reproduce the behavior
Create a Sale Order with items that have a positive number in stock. At this point listing all Sales Orders, hovering over this SO should show a Green icon with a label "Shippable". Next create a Shipment (this will decrease the inventory). List all Sale Orders, hover over the SO will show a Red Icon with label "Not shippable". Wich is true, because there are no more items left but because of the shipment itself.

**Comment by atm-maxime** [comment]
Hi !
At first I couldn't reproduce because of the workflow module which automatically set the order to "delivered" after shipment validation. But now I have reproduced.
There are several issues with this "shippable" function and I doubt we can fix it quickly without having a new way of doing things.

Currently this icon just checks for each line of each order (validated or in progress) that the quantity is below the real stock.

It doesn't check on shipments already made on the order. But also if the same product is on several lines. E.g. I created an order with 2 lines of the same product, quantity 8 for both. My product has 15 in stock, still the list says it's shippable !

Furthermore, it should also check on other orders to define if an order is shippable or not. That's why many changes are needed. And this is the kind of thing that will slow down the list...
